aws/logs_monitoring/lambda_function.py

```diff
@@ -390,14 +399,24 @@ def datadog_forwarder(event, context):
 
     if DD_FORWARD_LOG:
         forward_logs(filter_logs(map(json.dumps, logs)))
+        DD_FORWARDER_TELEMETRY["logs_forwarded"]+=len(list(filter_logs(map(json.dumps, logs))))
 
     forward_metrics(metrics)
+    DD_FORWARDER_TELEMETRY["metrics_forwarded"]+=len(metrics)
 
     if len(trace_payloads) > 0:
+        DD_FORWARDER_TELEMETRY["traces_forwarded"]+=len(trace_payloads)
         forward_traces(trace_payloads)
 
     parse_and_submit_enhanced_metrics(logs)
 
+    send_dd_forwarder_telemetry()
+
+    # Reset DD_FORWARDER_TELEMETRY counters and tags after sending
+    DD_FORWARDER_TELEMETRY["logs_forwarded"] = 0
```

Comment:
I don't think you need to keep a counter at all. Why not just submit the `len(metrics)` right away?

Reply:
Removed counter construction and submitting telemetry metrics right away (but still need to build the tags list)